lib/scientist/observation.rb

```diff
@@ -20,19 +20,29 @@ class Scientist::Observation
   # The Float seconds elapsed.
   attr_reader :duration
 
+  # The Float CPU time elapsed, in seconds
+  attr_reader :cpu_time
+
   def initialize(name, experiment, fabricated_duration: nil, &block)
     @name       = name
     @experiment = experiment
 
-    starting = Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_second) unless fabricated_duration
+    start_wall_time, start_cpu_time = capture_times unless fabricated_duration
+
     begin
       @value = block.call
     rescue *RESCUES => e
       @exception = e
     end
 
-    @duration = fabricated_duration ||
-      Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_second) - starting
+    if fabricated_duration
```

Comment:
This is not a backwards-compatible change. What do you think about keeping support for the old version (single value) alongside the new hash-based version?

Reply:
Hi Nathan 👋 

Thanks for the review, happy to make the change.

Reply:
I've pushed up the changes you requested! This is ready for a re-review.